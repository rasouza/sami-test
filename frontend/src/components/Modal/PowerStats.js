import React, { Fragment } from "react";
import { List, Header } from "semantic-ui-react"

export default (props) => {
  const { attributes } = props

  return (
    <Fragment>
      <Header>Power Stats</Header>
      <List selection verticalAlign="middle">
        <List.Item>
          <List.Header>Intelligence</List.Header>
          {attributes.intelligence}
        </List.Item>
        <List.Item>
          <List.Header>Strength</List.Header>
          {attributes.strength}
        </List.Item>
        <List.Item>
          <List.Header>Speed</List.Header>
          {attributes.speed}
        </List.Item>
        <List.Item>
          <List.Header>Durability</List.Header>
          {attributes.durability}
        </List.Item>
        <List.Item>
          <List.Header>Power</List.Header>
          {attributes.power}
        </List.Item>
        <List.Item>
          <List.Header>Combat</List.Header>
          {attributes.combat}
        </List.Item>
      </List>
    </Fragment>
  )
};