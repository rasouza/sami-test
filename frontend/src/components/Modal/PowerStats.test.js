import React from "react";
import { shallow } from "enzyme";
import PowerStats from "./PowerStats"
import { List } from "semantic-ui-react"

const attributes = {
  intelligence: "100",
  strength: "26",
  speed: "27",
  durability: "50",
  power: "47",
  combat: "100",
};

it('renders 6 attributes', () => {
  const actual = shallow(<PowerStats attributes={attributes} />)
  expect(actual.find(List.Item)).toHaveLength(6)
})

it('renders with default style', () => {
  const actual = shallow(<PowerStats attributes={attributes} />);
  expect(actual).toMatchSnapshot()
})