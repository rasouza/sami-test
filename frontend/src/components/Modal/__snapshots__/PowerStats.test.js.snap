// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with default style 1`] = `ShallowWrapper {}`;
